File: Assignment22_5.java
import java.util.Scanner;

////////////////////////////////////////////////////////////////////////////////
//    Function Name :       Frequency
//    Description :         To count frequency of given number in array
//    Input :               Integer
//    Output :              Integer
//    Author :              Raj Samir Jadhav
//    Date :                16/11/2025
////////////////////////////////////////////////////////////////////

public class FrequencyNumber
{
    public static int Frequency(int arr[], int length, int iNo)          // Function to getfrequency of iNo in array
    {
        int freq = 0;

        for(int i = 0; i < length; i++)                                 // Traverse array

        {
            if(arr[i] == iNo)   
            {
                freq++;                                             // Increase count
            }
        }
        return freq;                                               // Return final frequency
    }

    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter number of elements: ");            // Accept number of elements
        int size = sc.nextInt();

        System.out.print("Enter the number: ");                    // Accept number to search
        int value = sc.nextInt();

        int arr[] = new int[size];

        System.out.println("Enter " + size + " elements:");         // Accept elements
        for(int i = 0; i < size; i++)
        {
            System.out.print("Enter element " + (i + 1) + ": ");
            arr[i] = sc.nextInt();
        }

        int result = Frequency(arr, size, value);                    // Call function

        System.out.println("Frequency of " + value + " is: " + result);

        sc.close();
    }
}

File: Assignment22_5.c
#include <stdio.h>
#include <stdlib.h>

////////////////////////////////////////////////////////////////////////////////
//    Function Name :       Frequency
//    Description :         To count frequency of given number in array
//    Input :               Integer
//    Output :              Integer
//    Author :              Raj Samir Jadhav
//    Date :                16/11/2025
////////////////////////////////////////////////////////////////////

int Frequency(int Arr[], int iLength, int iNo)            // Function to count frequency of given number in array 
{
    int iCnt = 0, iFreq = 0;

    for(iCnt = 0; iCnt < iLength; iCnt++)                // Traverse the array
    {
        if(Arr[iCnt] == iNo)                            // Check if element matches given number
        {
            iFreq++;                                    // Increase frequency count
        }
    }
    return iFreq;                                     // Return final frequency
}

int main()
{
    int iSize = 0, iRet = 0, iCnt = 0, iValue = 0;
    int *p = NULL;

    printf("Enter number of elements: ");               // Accept number of elements
    scanf("%d", &iSize);

    printf("Enter the number: ");                        // Accept the number whose frequency is required
    scanf("%d", &iValue);

   
    p = (int *)malloc(iSize * sizeof(int));              // Allocate memory dynamically
    if(p == NULL)
    {
        printf("Unable to allocate memory\n");
        return -1;
    }

    printf("Enter %d elements:\n", iSize);              // Accept array elements
    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        printf("Enter element %d: ", iCnt + 1);
        scanf("%d", &p[iCnt]);
    }

    iRet = Frequency(p, iSize, iValue);                // Call function

    printf("Frequency of %d is: %d\n", iValue, iRet);  // Display result

    free(p);                                            // Free memory

    return 0;
}
